ui/static/ui/js/manage_taxonomies.jsx

```diff
@@ -6,24 +6,116 @@ define('manage_taxonomies', ['react', 'lodash', 'jquery', 'utils'],
   var ICheckbox = Utils.ICheckbox;
 
   var TermComponent = React.createClass({
+    mixins: [React.addons.LinkedStateMixin],
+    getInitialState: function() {
+      return {
+        formatActionClassName: 'fa fa-pencil',
+        formatActionState: 'edit',
+        editTextClass: 'form-control edit-term-box-hide',
+        listClassName: 'form-group',
+        label: ''
+      };
+    },
+    componentWillMount: function() {
+      if (this.props.term.label) {
+        this.setState({label: this.props.term.label});
+      }
+    },
     render: function () {
-      return <li>
+      var editBoxId = "edit_" + this.props.term.id;
+      return <li className={this.state.listClassName}>
         <span className="utility-features">
-          <a href="#">
-            <i className="fa fa-pencil"></i>
-          </a> <a href="#">
+          <a onClick={this.formatHandler} className="format-button">
+            <i className={this.state.formatActionClassName}></i>
+          </a> <a onClick={this.revertHandler} className="revert-button">
             <i className="fa fa-remove"></i>
           </a>
-        </span> <label className="term-title"
-        htmlFor="minimal-checkbox-1-11">{this.props.term.label}</label></li>;
+        </span>
+        <label className="term-title" id={this.props.term.id}
+        htmlFor="minimal-checkbox-1-11">{this.state.label}</label> <input
+        name="exit_term_temp" type="text" id={editBoxId}
+          valueLink={this.linkState('label')}
+          className={this.state.editTextClass}/> <label htmlFor={editBoxId}
+        className="help-inline control-label">{this.state.errorMessage}</label>
+      </li>;
+    },
+    formatHandler: function() {
+      var formatActionState = this.state.formatActionState;
+      var replaceWith = $('#edit_' + this.props.term.id);
+      var labelSelector = $('#' + this.props.term.id);
```

Comment:
The main problem with global selectors is that they may collide with other elements with the same id outside the react element. Since we can use the `ref` attribute to identify DOM elements instead, we don't have a need for `id` attributes here.

React may delete or modify DOM elements on any render so we should avoid touching the DOM as much as possible, instead using `state` to communicate these changes to the `render` method. This also makes the code cleaner since all rendering changes are in one place. Instead of calling `labelSelector.hide()` you should change the state using `setState` and update `render` to not display the label in that case.

For `replaceWith.focus()` you will need to access the node directly, try something like `$(React.findDOMNode(this.refs.exitTempTerm)).focus()`